File: java/brasil-burger/src/main/java/com/brasilburger/model/entities/Complement.java
package com.brasilburger.model.entities;

import com.brasilburger.core.interfaces.IEntity;

public class Complement implements IEntity {

    private int id;
    private String nom;
    private double prix;
    private String imageUrl;
    private String type; // boisson, frites, autre

    public Complement() {
    }

    public Complement(int id, String nom, double prix, String imageUrl, String type) {
        this.id = id;
        this.nom = nom;
        this.prix = prix;
        this.imageUrl = imageUrl;
        this.type = type;
    }

    // Getters & Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public double getPrix() {
        return prix;
    }

    public void setPrix(double prix) {
        this.prix = prix;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    @Override
    public String toString() {
        return nom + " (" + prix + " FCFA, " + type + ")";
    }
}

File: src/Entity/Complement.php
<?php
namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name:"Complements")]
class Complement
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(name:"Nom", type:"string", length:255)]
    private string $nom;

    #[ORM\Column(name:"Type", type:"string", length:50)]
    private string $type;

    #[ORM\Column(name:"Prix", type:"float")]
    private float $prix;

    #[ORM\Column(name:"ImageUrl", type:"string", length:255, nullable:true)]
    private ?string $imageUrl = null;

    #[ORM\Column(name:"EstDisponible", type:"boolean")]
    private bool $estDisponible = true;

    // getters et setters
    public function getId(): ?int { return $this->id; }
    public function getNom(): string { return $this->nom; }
    public function setNom(string $nom): self { $this->nom = $nom; return $this; }
    public function getType(): string { return $this->type; }
    public function setType(string $type): self { $this->type = $type; return $this; }
    public function getPrix(): float { return $this->prix; }
    public function setPrix(float $prix): self { $this->prix = $prix; return $this; }
    public function getImageUrl(): ?string { return $this->imageUrl; }
    public function setImageUrl(?string $imageUrl): self { $this->imageUrl = $imageUrl; return $this; }
    public function getEstDisponible(): bool { return $this->estDisponible; }
    public function setEstDisponible(bool $estDisponible): self { $this->estDisponible = $estDisponible; return $this; }
}
